### PR TITLE
replacing quadratic check with linear

### PR DESCRIPTION
**Pull request by joigno**
### Purpose for this PR

Close ticket #330, replacing the quadratic check of duplicated tokens for rewards with a linear check. Is minor but might appear on future audits.

It was passing all the tests and I added 1 more test for each contract I changed.

### CODE REVIEW

File: contracts/staking/StakingPool.sol
Comment (CjHare) [inline contracts/staking/StakingPool.sol:598]: suggestion: would it make sense to change the `revert` for a `require`?

_It may read easier as a require as we are not evaluating any complex expression i.e. it's a true or false_

**Review by CjHare — COMMENTED** [review]
Looking good, just the one question/suggestion

File: test/contracts/staking/staking.test.ts
Comment (joigno) [reply test/contracts/staking/staking.test.ts:184]: thanks, minted tokens and added allowance now!

File: contracts/staking/StakingPool.sol
Comment (joigno) [reply contracts/staking/StakingPool.sol:598]: thanks, it will be easier to read.

**Comment by joigno** [comment]
Added also `_enforceUniqueRewardTokens(_rewardTokens);` to `_initializeRewardTokens()`.

File: contracts/staking/StakingPool.sol
Comment (joigno) [reply contracts/staking/StakingPool.sol]: awaiting approval, thanks a lot.

**Review by CjHare — APPROVED** [review]
LGTM

**Comment by CjHare** [comment]
@joigno your first commit too 🎉 

_Turns out that first time committers need the workflow run approved, before it run the CI, which I've only just realised now_

**Comment by CjHare** [comment]
It's alright to merge 👍

**Pull request merged**
